src/include/sof/audio/component.h

```diff
@@ -46,13 +46,14 @@ struct dai_ts_data;
 /** \name Audio Component States
  *  @{
  */
-#define COMP_STATE_INIT		0	/**< Component being initialised */
-#define COMP_STATE_READY	1	/**< Component inactive, but ready */
-#define COMP_STATE_SUSPEND	2	/**< Component suspended */
-#define COMP_STATE_PREPARE	3	/**< Component prepared */
-#define COMP_STATE_PAUSED	4	/**< Component paused */
-#define COMP_STATE_ACTIVE	5	/**< Component active */
-#define COMP_STATE_PRE_ACTIVE	6	/**< Component after early initialisation */
+#define COMP_STATE_NOT_EXIST    0	/**< Component does not exist */
+#define COMP_STATE_INIT		1	/**< Component being initialised */
+#define COMP_STATE_READY	2	/**< Component inactive, but ready */
+#define COMP_STATE_SUSPEND	3	/**< Component suspended */
+#define COMP_STATE_PREPARE	4	/**< Component prepared */
+#define COMP_STATE_PAUSED	5	/**< Component paused */
+#define COMP_STATE_ACTIVE	6	/**< Component active */
+#define COMP_STATE_PRE_ACTIVE	7	/**< Component after early initialisation */
```

Comment:
@marcinszkudlinski  Any strong reason to reassign the numbers and not add NOT_EXIST at the end? I guess one possibility is some comparisons on state in existing code. I guess this is ok, but yeah, some of us have been staring at the raw component state numbers for years in log files and will need to relearn these new numbers, so not a small change.
UPDATE: there are quite a few "if state > READY" checks in existing code, has to be done like this.

Reply:
> UPDATE: there are quite a few "if state > READY" checks in existing code, has to be done like this.

yes, that was the reason